### DISCUSSION TITLE
vfmt should not touch blackslashed long lines

### DISCUSSION THREAD

**Discussion by blackshirt** [vlang/v#11819] [Ideas]
I have a silly issues with how `v fmt` works for formatting very long string and splitted to new line with a valid blackshlash `\`.
it's naturally easy to read and easy to grasp when long lines string splitted to new line with backslash.
i have the sample code here :
```v
fn (c CPool) update_single_rup(rup Rup) ?int {
	if c.rup_withkode_exist(rup.kode_rup) {
		q := "update Rup set nama_paket = '$rup.nama_paket', pagu = '$rup.pagu', \
				awal_pemilihan = '$rup.awal_pemilihan', metode = '$rup.metode', \
				sumber_dana = '$rup.sumber_dana', kegiatan = '$rup.kegiatan', \
				year = '$rup.year', last_updated = '$rup.last_updated', \
				tipe = '$rup.tipe' where kode_rup = '$rup.kode_rup';"
		_, code := c.exec(q)
		if code !in [0, 101] {
			return error(" Error in query exec with code : ${code}")
		}
		return code
	}
	return error("Rup ${rup.kode_rup} does not exist")
}
```

but when i run `v fmt` for this code, its removes the backslash and the `query` string becomes to very very long
```q := "update Rup set nama_paket = '$rup.nama_paket', pagu = '$rup.pagu', awal_pemilihan = '$rup.awal_pemilihan', metode = '$rup.metode', sumber_dana = '$rup.sumber_dana', kegiatan = '$rup.kegiatan', year = '$rup.year', last_updated = '$rup.last_updated', tipe = '$rup.tipe' where kode_rup = '$rup.kode_rup';"```

Its about 317 cols in my vscode editor, and look very awkward and really unreadable.

Just for suggestion, i think `v fmt` should not touchs manually defined blackslash for long line splitting, or automatically split long line string to multi line string with backlslash for more look easy reading and understandable code
Thanks

**Comment by serkonda7** [comment]
Right now you could achieve this by using string concatination as workaround:
```
q := "update Rup set nama_paket = '$rup.nama_paket', pagu = '$rup.pagu',"
				+ " awal_pemilihan = '$rup.awal_pemilihan', metode = '$rup.metode',"
				+ " sumber_dana = '$rup.sumber_dana', kegiatan = '$rup.kegiatan',"
				+ " year = '$rup.year', last_updated = '$rup.last_updated',"
				+ " tipe = '$rup.tipe' where kode_rup = '$rup.kode_rup';"
```

**Comment by blackshirt** [comment]
thanks ...its not a problem for simple thing, i can live with this 
but imho, maybe we suffered in classical problem when we concatenate a string, we are creating a new string, copying the contents of both the first and second argument into the new array. Needless to say, concatenation via the + operator is unnecessarily costly in most situations
